feat(notification): implement Android and iOS APIs

lucasfernog opened the pull request.

lucasfernog committed:
feat(notification): add plugin

lucasfernog committed:
move logic to init script

lucasfernog committed:
Merge branch 'next' into feat/notification

lucasfernog committed:
feat: add android impl

lucasfernog committed:
fix sdk

lucasfernog committed:
update api

lucasfernog committed:
ios impl

lucasfernog committed:
update android impl

lucasfernog committed:
nullable strings

lucasfernog committed:
lateinit vars

lucasfernog committed:
impl rust types

lucasfernog committed:
fix ios dates

lucasfernog committed:
rust and TS apis

lucasfernog committed:
simplify android impl

lucasfernog committed:
enhance errors

lucasfernog committed:
save notification

lucasfernog committed:
document silent option

lucasfernog committed:
let

lucasfernog committed:
add missing APIs

lucasfernog committed:
trigger events on android

lucasfernog committed:
trigger events

lucasfernog committed:
Merge remote-tracking branch 'origin/next' into feat/notification-mobile

FabianLars reviewed (approved):
(Take this review with a grain of salt since i can't test it, but it _looks_ good :D )

lucasfernog commented:
Heavily inspired by https://github.com/ionic-team/capacitor-plugins/tree/main/local-notifications/android/src/main/java/com/capacitorjs/plugins/localnotifications (I still need to find out where I should put this info). I basically took their implementation and refactored it, fixed some TODOs and changed the API to work better with our Rust backend (that's why you'll see an "enum with associated data" in Kotlin lol).

lucasfernog committed:
load config

lucasfernog committed:
Merge branch 'v2' into feat/notification-mobile
